### ISSUE TITLE
validation step crashes with TypeError: unhashable type: 'numpy.ndarray'

### ISSUE THREAD

**Issue by SamuelBrudner** [NickleDave/vak#239]
I am attempting to isolate bouts of singing (not individuate syllables). I converted praat annotations to .not.mat files with two labels: 'b' for timepoints where the animal is singing a bout, and '-' for everything else (silence, cage noise, calling without singing). "vak train" runs through several training epochs, but throws an error when it evaluates the validation set for the first time. I copied the files that generate this error to the 'boutSegmentation' folder of 'getVakWorkingForSam' -- my shared google drive folder with David. The error message is below.

![mooneyUbuntu__mooneyUbuntu__-_VNC_Viewer](https://user-images.githubusercontent.com/8064567/85580540-d4d65a80-b609-11ea-9328-cf309711635d.png)

**Comment by NickleDave** [comment]
Hi @SamuelBrudner thank you for reporting this issue. I'm sorry I haven't got back to you sooner.

I think there must be something wrong with `inverse_labels_mapping` but I haven't figured out why yet.

I'm really swamped this week, but I will find a way to debug this with the files you uploaded by the end of this weekend at the latest.

**Comment by NickleDave** [comment]
Just confirming: I got the same error.
Dumping  the last bit of the traceback here in text just so it's searchable in case anyone else has a similar error.
```
    load_entry_point('vak', 'console_scripts', 'vak')()
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/__main__.py", line 44, in main
    config_file=args.configfile)
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/cli/cli.py", line 22, in cli
    train(toml_path=config_file)
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/cli/train.py", line 71, in train
    logger=logger,
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/core/train.py", line 289, in train
    device=device)
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/engine/model.py", line 415, in fit
    ckpt_step)
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/engine/model.py", line 128, in _train
    metric_vals = self._eval(val_data)
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/engine/model.py", line 237, in _eval
    y_labels = lbl_tb2labels(y.cpu().numpy(), eval_data.dataset.labelmap)
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/labels.py", line 206, in lbl_tb2labels
    labels = [inverse_labels_mapping[label] for label in labels]
  File "/home/art/Documents/repos/coding/birdsong/vak/src/vak/labels.py", line 206, in <listcomp>
    labels = [inverse_labels_mapping[label] for label in labels]
TypeError: unhashable type: 'numpy.ndarray'

Seeing if I can figure out what's going on now 
```

**Comment by NickleDave** [comment]
@SamuelBrudner I merged a branch in that should fix this.
I was able to successfully train a model with your data to completion after adding these commits.

I think part of the reason we missed this bug was because we hadn't worked with data without clear "unlabeled silent" segments before--there was a logical error that depended in part on that. 
Sorry you had to be the one to find it, but thank you for helping fix it.

Hope this can get you going, feel free to raise another issue if you run into anything else though
